libs/application/data-providers/src/providers/PaymentCatalogProvider.ts

```diff
@@ -45,39 +45,34 @@ export class PaymentCatalogProvider extends BasicDataProvider {
 
     return this.useGraphqlGateway(query, {
       input: { performingOrganizationID: '6509142520' },
-    })
-      .then(async (res: Response) => {
-        // TODO: needs refactoring
-        const response = await res.json()
-        if (response.errors) {
-          return this.handleError()
-        }
+    }).then(async (res: Response) => {
+      // TODO: needs refactoring
+      const response = await res.json()
+      if (response.errors) {
+        return Promise.reject(response.errors)
+      }
 
-        if (response.data.paymentCatalog !== '') {
-          const correctCatalog = searchCorrectCatalog(
-            CHARGE_ITEM_CODE,
-            JSON.stringify(response.data.paymentCatalog.items),
-          )
+      if (response.data.paymentCatalog !== '') {
+        const correctCatalog = searchCorrectCatalog(
+          CHARGE_ITEM_CODE,
+          JSON.stringify(response.data.paymentCatalog.items),
+        )
 
-          if (correctCatalog !== '') {
-            return Promise.resolve(correctCatalog)
-          }
+        if (correctCatalog !== '') {
+          return Promise.resolve(correctCatalog)
+        } else {
+          return Promise.reject(response.data.paymentCatalog)
         }
-        return Promise.resolve(response.data.paymentCatalog.items)
-      })
-      .catch(() => {
-        return this.handleError()
-      })
-  }
+      }
 
-  handleError() {
-    return Promise.resolve({})
+      return Promise.resolve(response.data.paymentCatalog.items)
+    })
   }
 
-  onProvideError(result: string): FailedDataProviderResult {
+  onProvideError(result: any): FailedDataProviderResult {
```

Comment:
Why `any` rather than f.x. unknown or `DataProviderResult['data']`?

Reply:
really good point.. this was unintended. ill fix

Reply:
should be fixed now